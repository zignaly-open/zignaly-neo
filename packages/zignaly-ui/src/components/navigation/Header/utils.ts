// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
