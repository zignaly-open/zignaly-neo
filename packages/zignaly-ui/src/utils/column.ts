import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
