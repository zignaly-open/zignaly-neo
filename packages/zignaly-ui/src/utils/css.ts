import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
