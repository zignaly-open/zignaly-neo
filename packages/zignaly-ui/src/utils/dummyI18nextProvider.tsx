import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
