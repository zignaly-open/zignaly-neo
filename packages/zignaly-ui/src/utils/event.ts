export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
