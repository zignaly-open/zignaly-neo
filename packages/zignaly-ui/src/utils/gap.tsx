import styled from "styled-components";

export const Gap = styled.div<{ gap: number }>`
  ${(props) => `
  padding: ${props.gap}px;
  `}
`;
