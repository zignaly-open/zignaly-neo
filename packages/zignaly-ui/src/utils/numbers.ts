export const sortByValue = (number1: string | number, number2: string | number) => {
  return +number1 - +number2;
};

export const trimZeros = (v: string | number) => {
  return v?.toString()?.replace(/(\.\d*?[1-9]+)0+$|\.0+$/g, "$1");
};

export const numberOfDecimals = (v: string | number) => {
  return (v?.toString()?.split(".")[1] ?? "").length;
};

export const formatCompactNumber = (value: number | string, precision = 8) =>
  new Intl.NumberFormat("en-US", {
    maximumFractionDigits: precision,
    notation: "compact",
    compactDisplay: "short",
  }).format(+value);
