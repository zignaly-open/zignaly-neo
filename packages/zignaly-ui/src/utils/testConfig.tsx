import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
