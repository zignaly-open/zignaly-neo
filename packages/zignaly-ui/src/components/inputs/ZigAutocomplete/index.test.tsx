import React from "react";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { fireEvent, screen, waitFor } from "@testing-library/react";
import ZigAutocomplete from ".";

describe("components/ZigAutocomplete", () => {
  const options = ["Option 1", "Option 2", "Option 3"];

  it("should render without crashing", () => {
    const { container } = renderWithProvidersUi(<ZigAutocomplete id="test-id" options={options} />);

    const autocomplete = container.querySelector("#test-id") as HTMLElement;
    expect(autocomplete).toBeVisible();
    expect(autocomplete).toMatchSnapshot();
  });

  it("should open dropdown on focus", () => {
    renderWithProvidersUi(<ZigAutocomplete id="test-id" options={options} />);

    const input = screen.getByRole("combobox");
    fireEvent.focus(input);

    const listItem = screen.getByText(options[0]);
    expect(listItem).toBeVisible();
  });

  it("should select item on click", () => {
    const handleChange = jest.fn();
    renderWithProvidersUi(
      <ZigAutocomplete id="test-id" options={options} onChange={handleChange} />,
    );

    const input = screen.getByRole("combobox") as HTMLInputElement;
    fireEvent.focus(input);

    const listItem = screen.getByText(options[0]);
    fireEvent.click(listItem);

    waitFor(() => {
      expect(handleChange).toHaveBeenCalled();
      expect(input.value).toBe(options[0]);
    });
  });

  it("should display error message when `error` prop is passed", async () => {
    renderWithProvidersUi(<ZigAutocomplete id="test-id" error="Test error" options={options} />);

    const input = screen.getByRole("combobox") as HTMLInputElement;
    fireEvent.focus(input);

    await waitFor(() => {
      expect(screen.getByText("Test error")).toBeInTheDocument();
    });
  });
});
