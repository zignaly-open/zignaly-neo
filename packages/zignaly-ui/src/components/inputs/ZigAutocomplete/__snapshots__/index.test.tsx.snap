// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ZigAutocomplete should render without crashing 1`] = `
<input
  aria-autocomplete="list"
  aria-expanded="false"
  aria-invalid="false"
  autocapitalize="none"
  autocomplete="off"
  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1fsdrue-MuiInputBase-input-MuiInput-input"
  data-testid="test-id"
  id="test-id"
  role="combobox"
  spellcheck="false"
  type="text"
  value=""
/>
`;
